docs/howto/query.md

```diff
@@ -0,0 +1,227 @@
+(query)=
+# How to query the database
+
+## Create/Make your types usable by SQLair
+The first step in building your query is to decide what you would like to put
+into/get out of the database. Identify the types in your Go program that hold
+this information or create them if they don't already exist.
+
+Make sure that the types are in the right format to work with SQLair.
+
+For example:
+```go
+type Employee struct {
+    ID         string `db:"id"`
+    Name       string `db:"name"`
+    LocationID string `db:"location_id"`
+}
+
+type Location struct {
+    ID   string `db:"id"`
+    City string `db:"city"`
+}
+```
+
+> See more: {ref}`types` 
+
+## Write your query string using the types
+
+SQLair queries are regular SQL queries with special input and output expressions
+inserted which allow you to reference the Go types directly from the SQL.
+
+Write your SQL query with SQLair input expressions instead of query parameters
+and SQLair output expressions instead of columns to select.
+
+For example:
+```
+query := `
+    SELECT &Employee.*
+    FROM employee
+    WHERE location_id = $Location.id
+`
+```
+
+> See more {ref}`input-expression-syntax`, {ref}`output-expression-syntax`.
+
+## Prepare your query string to get a Statement
+
+Now you have your query string, you need to pass it to `sqlair.Prepare` along
+with all the types mentioned in the input and output expressions. 
+
+SQLair uses the type information to generate the SQL and verify the input/output
+expressions. If there is an issue with the actual SQL in the query, this will
+not catch it. The returned `Statement` holds the parsed and verified query.
+
+
+For example:
+```go
+stmt, err := sqlair.Prepare(ctx, query, Employee{}, Location{})
+if err != nil {
+    return err
+}
+```
+
+> See more [`sqlair.Preapre`](https://pkg.go.dev/github.com/canonical/sqlair#Prepare)
+
+## Execute the Statement on the database
+
+Execute the statement on a `DB` or a `TX` using the `Query` method on either.
+`Query` takes the `Statement` and all the input arguments specified in the input
+expressions. This returns a `Query` object that can then be run with one of four
+methods below.
+
+```{note}
+The `Query` object returned from `DB.Query`/`TX.Query` is not designed to be
+reused. One of the methods on `Query` should immidetly be called. It should not
+be saved as variable.
+```
+
+> See more:
+[`DB.Query`](https://pkg.go.dev/github.com/canonical/sqlair#DB.Query),
+[`TX.Query`](https://pkg.go.dev/github.com/canonical/sqlair#TX.Query),
+[`sqlair.Query`](https://pkg.go.dev/github.com/canonical/sqlair#Query)
+
+### Get one row
+To get only the first row returned from the database, use `Query.Get`. Pointers to
+all the output variables mentioned in the query should be passed to `Query.Get`.
```

Comment:
Restate the second sentence so it's active (as in, don't tell me what should be done in the abstract, tell me what to do now -- this is a how-to guide). E.g.,

>  use `Query.Get(), passing as attribute pointers to all the output variables mentioned...`